endpoints/music_endpoint.py

```diff
@@ -56,7 +57,13 @@ def post(self, request, room, format=None):
             required: true
         """
         request.data.update({'room_id': room.id})
-        serializer = MusicSerializer(data=request.data)
+        try:
+            music = Music.objects.get(music_id=request.data.get('music_id'), room=room)
+            serializer = MusicSerializer(music, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+        except Music.DoesNotExist:
+            serializer = MusicSerializer(data=request.data)
         if serializer.is_valid():
```

Comment:
If music already exists and is just updated, you check twice isValid(), re-use add_music() that check if the music already exists too. We should simplify the behavior